libraries/node-core-library/src/FileSystem.ts

```diff
@@ -829,25 +892,98 @@ export class FileSystem {
    * @param linkPath - The path to the link.
    */
   public static getRealPath(linkPath: string): string {
-    return fsx.realpathSync(linkPath);
+    return FileSystem._wrapException(() => {
+      return fsx.realpathSync(linkPath);
+    });
   }
 
   /**
    * An async version of {@link FileSystem.getRealPath}.
    */
   public static getRealPathAsync(linkPath: string): Promise<string> {
-    return fsx.realpath(linkPath);
+    return FileSystem._wrapExceptionAsync(() => {
+      return fsx.realpath(linkPath);
+    });
   }
 
   // ===============
   // UTILITY FUNCTIONS
   // ===============
 
   /**
-   * Returns true if the error provided indicates the file or folder
-   * does not exist.
+   * Returns true if the error provided indicates the file or folder does not exist.
    */
   public static isNotExistError(error: NodeJS.ErrnoException): boolean {
-    return error.code === 'ENOENT' || error.code === 'ENOTDIR';
+    return FileSystem.isFileDoesNotExistError(error) || FileSystem.isFolderDoesNotExistError(error);
+  }
+
+  /**
+   * Returns true if the error provided indicates the file does not exist.
+   */
+  public static isFileDoesNotExistError(error: NodeJS.ErrnoException): boolean {
+    return FileSystem._isErrorType(error) && (error.code === 'ENOENT');
+  }
+
+  /**
+   * Returns true if the error provided indicates the folder does not exist.
+   */
+  public static isFolderDoesNotExistError(error: NodeJS.ErrnoException): boolean {
+    return FileSystem._isErrorType(error) && (error.code === 'ENOTDIR');
+  }
+
+  private static _isErrorType(error: NodeJS.ErrnoException): boolean {
```

Comment:
`NodeJS.ErrnoException` is not the correct type, given how these functions are used.  The type should be `Error`.  (The actual `catch` parameter has type `any` and can even be weird values such as `string` or `boolean`, but our ESLint rules and coding conventions mandate that it must be an object implementing the `Error` contract.)